experimenter/experimenter/outcomes/__init__.py

```diff
@@ -91,6 +91,13 @@ def all(cls):
     def by_application(cls, application):
         return [o for o in cls.all() if o.application == application]
 
+    @classmethod
+    def get_by_slug(cls, slug):
+        for o in cls.all():
+            if o.slug == slug:
+                return o
```

Comment:
A small thing but pretty much always avoid using single letter variable names, the only time i would use them is within a single comprehension ie `[o for o in outcomes]` but here it should be `outcome`